src/utils/templates.ts

```diff
@@ -20,28 +23,32 @@ const httpMethods = [
   'trace',
   'patch',
 ]
-const api: Template = ({ name, args }) => ({
-  path: `server/api/${name}${applySuffix(args, httpMethods, 'method')}.ts`,
-  contents: `
+const api: Template = ({ name, args, nuxt }) => {
+  const pathRoot = nuxt.future.compatibilityVersion === 4 ? nuxt.rootDir : nuxt.srcDir
+
+  return {
+    path: resolve(pathRoot, `${nuxt.serverDir}/api/${name}${applySuffix(args, httpMethods, 'method')}.ts`),
+    contents: `
```

Comment:
```suggestion
  return {
    path: resolve(nuxt.srcDir, nuxt.serverDir, `api/${name}${applySuffix(args, httpMethods, 'method')}.ts`),
    contents: `
```

Reply:
> ... the serverDir is absolute ...

Ah I see, I thought it was a relative path!